src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1625,6 +1658,9 @@ static int add_object_vtable_internal(
         const sd_bus_vtable *v;
         struct node *n;
         int r;
+        const char *names;
+        int names_flags;
```

Comment:
why `int`? just make the enum you defined above a named enum type, then use that type here.

Reply:
this would be incorrect because the enum is a bitfield, so a combination first|have_names doesn't belong to it. I will trade the enum for a struct.

Reply:
storing bitfields in enum is totally ok. When debugging such values with gdb it is even smart enough to decode the bitfields for you. Hence, please use a properly typed enum here, it makes things a ton more readable.

Reply:
ok, I will keep the enum, but I remain convinced that it is only half ok. In a switch(myenumval), it will never jump to a composite value. gdb can be smart, but here we are bending the rules.